immut/sorted_map/utils.mbt

```diff
@@ -257,3 +257,13 @@ pub impl[K : Show, V : Show] Show for T[K, V] with output(self, logger) {
 pub fn to_string[K : Show, V : Show](self : T[K, V]) -> String {
   Show::to_string(self)
 }
+
+pub impl[K : Show, V : ToJson] ToJson for T[K, V] with to_json(self) {
```

Comment:
maybe pre-alloc memory with capacity specified? As we've already know the size of the `jsons`. 

Same applies for other impl

Reply:
@qazxcdswe123 Addressed for `sorted_map`. However, lengths of `list`s and `sorted_set`s are (currently) not pre-calculated AFAIK, so it's probably out of the scope of this PR to make it so.

Reply:
@qazxcdswe123 Addressed for `list`s and `sorted_set`s by calculating the capacity required ad hoc.